WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderDetailHeaderView.swift

```diff
@@ -0,0 +1,326 @@
+
+        // Enable button only if not previewing a site.
+        if let topic = post?.topic {
+            blogNameButton.isEnabled = !ReaderHelpers.isTopicSite(topic)
+        }
+
+        // If the button is enabled also listen for taps on the avatar.
+        if blogNameButton.isEnabled {
+            let tgr = UITapGestureRecognizer(target: self, action: #selector(didTapHeaderAvatar(_:)))
+            blavatarImageView.addGestureRecognizer(tgr)
+        }
+    }
+
+    private func configureFeaturedImage() {
+        guard let post = post,
+            !post.contentIncludesFeaturedImage(),
+            let featuredImageURL = post.featuredImageURLForDisplay() else {
+                return
+        }
+
+        let host = MediaHost(with: post, failure: { error in
+            // We'll log the error, so we know it's there, but we won't halt execution.
+            CrashLogging.logError(error)
+        })
+
+        let maxImageWidth = frame.width
+        let imageWidthSize = CGSize(width: maxImageWidth, height: 0) // height 0: preserves aspect ratio.
+        featuredImageLoader.loadImage(with: featuredImageURL, from: host, preferredSize: imageWidthSize, placeholder: nil, success: { [weak self] in
+            guard let strongSelf = self, let size = strongSelf.featuredImageView.image?.size else {
+                return
+            }
+            DispatchQueue.main.async {
+                strongSelf.configureFeaturedImageConstraints(with: size)
+                strongSelf.configureFeaturedImageGestures()
+            }
+        }) { error in
+            DDLogError("Error loading featured image in reader detail: \(String(describing: error))")
+        }
+    }
+
+    private func configureFeaturedImageConstraints(with size: CGSize) {
+        // Unhide the views
+        featuredImageView.isHidden = false
+        featuredImageBottomPaddingView.isHidden = false
+
+        // Now that we have the image, create an aspect ratio constraint for
+        // the featuredImageView
+        let ratio = size.height / size.width
+        let constraint = NSLayoutConstraint(item: featuredImageView as Any,
+                                            attribute: .height,
+                                            relatedBy: .equal,
+                                            toItem: featuredImageView!,
+                                            attribute: .width,
+                                            multiplier: ratio,
+                                            constant: 0)
+        constraint.priority = .defaultHigh
+        featuredImageView.addConstraint(constraint)
+        featuredImageView.setNeedsUpdateConstraints()
+    }
+
+    private func configureFeaturedImageGestures() {
+        // Listen for taps so we can display the image detail
+        let tgr = UITapGestureRecognizer(target: self, action: #selector(didTapFeaturedImage(_:)))
+        featuredImageView.addGestureRecognizer(tgr)
+
+        layoutIfNeeded()
```

Comment:
Why do we need to call `layoutIfNeeded` here?

Reply:
Honestly, I don't know. It was in the original method here: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift#L723

I removed it, and everything works fine.